pandas/tseries/frequencies.py

```diff
@@ -233,6 +233,7 @@ def get_offset(name):
 
 
 def infer_freq(index, warn=True):
+    # type: (Union[Series, Index, PeriodArray]) -> Freq
```

Comment:
If you're OK with it, I'm going to copy this and a other non-central things into a hodge-podge PR; the smaller diff here should make review easier.